src/rez/package_order.py

```diff
@@ -608,6 +609,311 @@ def from_pod(cls, data):
+                default_key = priority_sort_key
+                continue
+            if version_range.contains_version(version):
+                break
+        else:
+            # For now, we're permissive with the version_sort_order - it may
+            # contain ranges which match no actual versions, and if an actual
+            # version matches no entry in the version_sort_order, it is simply
+            # placed after other entries
+            priority_sort_key = default_key
+
+        if self.version_orderer:
+            version_key = self.version_orderer.sort_key_implementation(package_name, version)
+        else:
+            version_key = version
+
+        return priority_sort_key, version_key
+
+    @staticmethod
+    def _packages_to_pod(packages: Dict[str, List[Union[str, VersionRange]]]) -> Dict[str, List[str]]:
+        return {
+            package: [str(v) for v in versions]
+            for (package, versions) in packages.items()
+        }
+
+    @staticmethod
+    def _packages_from_pod(
+            packages: Dict[str, List[Union[str, VersionRange]]],
+    ) -> Dict[str, List[Union[str, VersionRange]]]:
+        parsed_dict = {}
+        for package, versions in packages.items():
+            new_versions = []
+            num_false = 0
+            for v in versions:
+                if v in ("", False):
+                    v = False
+                    num_false += 1
+                else:
+                    if not isinstance(v, VersionRange):
+                        if isinstance(v, (int, float)):
+                            v = str(v)
+                        v = VersionRange(v)
+                new_versions.append(v)
+            if num_false > 1:
+                raise ConfigurationError("version_priorities for CustomPackageOrder may "
+                                         "only have one False / empty value")
+            parsed_dict[package] = new_versions
+        return parsed_dict
+
+    def to_pod(self):
+        return dict(
+            packages=self._packages_to_pod(self.packages_dict),
+            version_orderer=to_pod(self.version_orderer) if self.version_orderer else None,
+        )
+
+    @classmethod
+    def from_pod(cls, data):
+        version_orderer = data["version_orderer"]
+        return cls(
+            packages=data["packages"],
+            version_orderer=from_pod(version_orderer) if version_orderer else None,
+        )
+
+
+class PyPAPackageOrder(PackageOrder):
+    """
+    Here's how this package orderer behaves:
+    1. First separate versions into two groups based on the value of
+       ``prerelease``:  a disallowed/non-preferential group and those which are allowed/preferred
+    2. Sort each group from step 1 in the order defined by the pep440 spec.
+        e.g.  1.1 > 1.1rc1 > 1.1b1 > 1.1a1
+    3. Concatenate the two groups: placing the non-preferred group -- in sorted order --
+        after the preferred group.
```

Comment:
Hopefully the above example with `pip install` helps clarify it.
Essentially, most of the time you want users to get the latest release version, not the absolute latest version.  But you also don't want to specifically disallow them from using it.  The prerelease argument allows the package sorter to set a risk tolerance for the consumer, so that if I am comfortable with `rc`, but not `beta` then I am happy with the highest version number of either rc or release, but not beta or alpha.